subiquitycore/core.py

```diff
@@ -398,7 +398,8 @@ def next(loop, user_data):
 
     def run(self):
         if not hasattr(self, 'loop'):
-            if self.common['opts'].run_on_serial:
+            if (self.common['opts'].run_on_serial and
+                    os.environ["TERM"] != "vt220"):
```

Comment:
Why are you hard-coding vt220 as a whitelist for colors?  I think the reasonable default assumption today is that serial console DOES support colors, and it should only be the serial console on z systems, where the serial connection terminates on a java client that the end user has no control over, should be treated as monochrome.

Reply:
At the moment `--serial` is passed to subiquity on consoles that systemd generates / determines to be serial. And all of those end up without colors, on all arches.

You can see a good summary of console names here https://www.ibm.com/support/knowledgecenter/en/linuxonibm/com.ibm.linux.z.ludd/ludd_r_console_assure_nodes.html

Note, all of them are detected as serial, but only the `/dev/ttysclp0` is known to support colors, and that one is a `TERM=vt220`. (also kvm is not listed on that page, but `/dev/ttysclp0` is available there as the only console)

So how should this be done?

Should I add a `--force-color` flag to subiquity and then bake this flag as an override on the `serial-getty@ttysclp0.service`? or like add code in subiquity to specifically check for tty names?

We already have a `is_linux_tty` function, do you think that can be used to detect colorful serial consoles? https://github.com/CanonicalLtd/subiquity/blob/master/subiquitycore/core.py#L73

Reply:
(See other comment)